rust/src/lib.rs

```diff
@@ -695,1508 +323,454 @@ impl PartialEq for TransactionOutput {
+        data.as_slice().try_into().map(Self).map_err(|_e| {
+            let cbor_error = cbor_event::Error::WrongLen(
+                16,
+                cbor_event::Len::Len(data.len() as u64),
+                "Ipv6 address length",
+            );
+            DeserializeError::new("Ipv6", DeserializeFailure::CBOR(cbor_error))
+        })
     }
 
-    pub fn has_script_credentials(&self) -> bool {
-        self.stake_credential.has_script_hash()
+    pub fn ip(&self) -> Vec<u8> {
+        self.0.to_vec()
     }
 }
 
+static URL_MAX_LEN: usize = 128;
+
 #[wasm_bindgen]
 #[derive(
-    Clone, Debug, Hash, Eq, Ord, PartialEq, PartialOrd, serde::Serialize, serde::Deserialize, JsonSchema,
+    Clone,
+    Debug,
+    Hash,
+    Eq,
+    Ord,
+    PartialEq,
+    PartialOrd,
+    serde::Serialize,
+    serde::Deserialize,
+    JsonSchema,
 )]
-pub struct Ed25519KeyHashes(pub(crate) Vec<Ed25519KeyHash>);
+pub struct URL(String);
 
-impl_to_from!(Ed25519KeyHashes);
+impl_to_from!(URL);
 
 #[wasm_bindgen]
-impl Ed25519KeyHashes {
-    pub fn new() -> Self {
-        Self(Vec::new())
-    }
-
-    pub fn len(&self) -> usize {
-        self.0.len()
-    }
-
-    pub fn get(&self, index: usize) -> Ed25519KeyHash {
-        self.0[index].clone()
-    }
-
-    pub fn add(&mut self, elem: &Ed25519KeyHash) {
-        self.0.push(elem.clone());
+impl URL {
+    pub fn new(url: String) -> Result<URL, JsError> {
+        Self::new_impl(url).map_err(|e| JsError::from_str(&e.to_string()))
     }
 
-    pub fn to_option(&self) -> Option<Ed25519KeyHashes> {
-        if self.len() > 0 {
-            Some(self.clone())
+    pub(crate) fn new_impl(url: String) -> Result<URL, DeserializeError> {
+        if url.len() <= URL_MAX_LEN {
+            Ok(Self(url))
         } else {
-            None
+            Err(DeserializeError::new(
+                "URL",
+                DeserializeFailure::OutOfRange {
+                    min: 0,
+                    max: URL_MAX_LEN,
+                    found: url.len(),
+                },
+            ))
         }
     }
-}
 
-impl IntoIterator for Ed25519KeyHashes {
-    type Item = Ed25519KeyHash;
-    type IntoIter = std::vec::IntoIter<Self::Item>;
-
-    fn into_iter(self) -> Self::IntoIter {
-        self.0.into_iter()
+    pub fn url(&self) -> String {
+        self.0.clone()
     }
 }
 
-#[wasm_bindgen]
-#[derive(
-    Clone, Debug, Hash, Eq, Ord, PartialEq, PartialOrd, serde::Serialize, serde::Deserialize, JsonSchema,
-)]
-pub struct Relays(Vec<Relay>);
-
-impl_to_from!(Relays);
-
-#[wasm_bindgen]
-impl Relays {
-    pub fn new() -> Self {
-        Self(Vec::new())
-    }
-
-    pub fn len(&self) -> usize {
-        self.0.len()
-    }
-
-    pub fn get(&self, index: usize) -> Relay {
-        self.0[index].clone()
-    }
-
-    pub fn add(&mut self, elem: &Relay) {
-        self.0.push(elem.clone());
-    }
-}
+static DNS_NAME_MAX_LEN: usize = 64;
```

Comment:
Shouldn't this be 128, because Conway CDDL says `dns_name` can be upto 128 chars long.
Ref: https://github.com/IntersectMBO/cardano-ledger/blob/8fd7ab6ca9bcf9cdb1fa6f4059f84585a084efa5/eras/conway/impl/cddl-files/conway.cddl#L354

Reply:
Thanks @itsfarseen ! We have updated it